### PR TITLE
Migrating dev server to Vite

### PR DESCRIPTION
**Pull request by pedrorezende**
We've been struggling with the lack of hot module replacement in several moments of the development. I've tried to tweak / change the Webpack config without much success - it kept reloading the whole structure on every change. 

So I took the first step to definitely solve this issue: migrate from Webpack to Vite. I would really appreciate your comments and thoughts about this.

#722

### CODE REVIEW

File: apps/namada-interface/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import react from "@vitejs/plugin-react";
+import { defineConfig } from "vite";
+import tsconfigPaths from "vite-tsconfig-paths";
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react(), tsconfigPaths()],
+  define: {
+    "process.env": {},
```
Comment (mateuszjasiuk) [inline apps/namada-interface/vite.config.js:9]: Is this enough to load envs?

File: apps/namada-interface/package.json [hunk 2]
```diff
@@ -48,7 +48,7 @@
     "watch": "webpack-cli --watch",
     "start": "webpack-dev-server",
```
Comment (mateuszjasiuk) [inline apps/namada-interface/package.json:49]: You can remove this  I guess :)

**Review by mateuszjasiuk — COMMENTED** [review]
👋 I think if it improves overall dev experience then we should go for it. It does not seem to work for me though. I'm getting:
`Uncaught SyntaxError: The requested module '/@fs/home/mj/Projects/heliax/namada-interface/packages/shared/src/shared/snippets/shared-801e3fcea73932a3/src/sdk/mod.js?t=1712224447927' does not provide an export named 'fetchAndStoreMaspParams' (at shared.js:2:40)`
When opening interface :(

File: apps/namada-interface/vite.config.js (hunk 1 above)
Comment (pedrorezende) [reply apps/namada-interface/vite.config.js:9]: It was a temp fix that I left behind :/ Checking that now

**Comment by pedrorezende** [comment]
> 👋 I think if it improves overall dev experience then we should go for it. It does not seem to work for me though. I'm getting: `Uncaught SyntaxError: The requested module '/@fs/home/mj/Projects/heliax/namada-interface/packages/shared/src/shared/snippets/shared-801e3fcea73932a3/src/sdk/mod.js?t=1712224447927' does not provide an export named 'fetchAndStoreMaspParams' (at shared.js:2:40)` When opening interface :(

Will make a clean install to try to reproduce this error o/

**Pull request merged**